pkg/policy/api/ingress.go

```diff
@@ -108,12 +108,35 @@ type IngressRule struct {
 	//
 	// +optional
 	FromEntities EntitySlice `json:"fromEntities,omitempty"`
+
+	aggregatedSelectors EndpointSelectorSlice
+}
+
+// SetAggregatedSelectors creates a single slice containing all of the following
+// fields within the IngressRule, converted to EndpointSelector, to be stored
+// within the IngressRule for easy lookup while performing policy evaluation
+// for the rule:
+// * FromEntities
+// * FromCIDR
+// * FromCIDRSet
+//
+// FromEndpoints is not aggregated due to requirement folding in
+// GetSourceEndpointSelectorsWithRequirements()
+func (i *IngressRule) SetAggregatedSelectors() {
+	res := make(EndpointSelectorSlice, 0, len(i.FromEndpoints)+len(i.FromEntities)+len(i.FromCIDRSet)+len(i.FromCIDR))
```

Comment:
If `FromEndpoints` is not being used why the `len(i.FromEndpoints) here

Reply:
Fixed.